### PR TITLE
Building names and counts added to buildings menu

### PR DESCRIPTION
**Pull request by tom-pratt**
Implemented building counts but has the same problem as the desktop version. Building counts are only updated when you move position on the map. Could work around by updating on every draw, but would be doing unnecessary work.

### CODE REVIEW

**Comment by Tom-Pratt** [comment]
screenshot_20170602-193649

**Comment by andreas-eberle** [comment]
Could you update it every second or every 30th draw?

File: jsettlers.main.android/src/main/java/jsettlers/main/android/core/controls/ControlsAdapter.java [hunk 1]
```diff
@@ -190,4 +208,35 @@ public boolean isTaskActive() {
...
 		androidControls.endTask();
 	}
+
+
+	/**
+	 * ParitionControls inplementation
+	 */
+	@Override
+	public boolean isInPlayerPartition() {
+		if (displayCenter == null)
```
Comment (andreas-eberle) [inline jsettlers.main.android/src/main/java/jsettlers/main/android/core/controls/ControlsAdapter.java]: please use { and }

File: jsettlers.main.android/src/main/java/jsettlers/main/android/core/controls/ControlsAdapter.java [hunk 2]
```diff
@@ -103,6 +111,16 @@ public void onDraw() {
 		}
 	}
 
+	public void onPositionChanged(MapRectangle screenArea, ShortPoint2D displayCenter) {
+		this.displayCenter = displayCenter;
+
+		synchronized (positionChangedListeners) {
+			for (PositionChangedListener listener : positionChangedListeners) {
```
Comment (andreas-eberle) [inline jsettlers.main.android/src/main/java/jsettlers/main/android/core/controls/ControlsAdapter.java]: use stream(..).forEach(..) here

File: jsettlers.main.android/src/main/java/jsettlers/main/android/core/controls/AndroidControls.java [hunk 3]
```diff
@@ -142,6 +142,11 @@ public String getDescriptionFor(UIPoint position) {
 
 	@Override
 	public void setMapViewport(MapRectangle screenArea) {
+		ShortPoint2D displayCenter = new ShortPoint2D(screenArea.getLineStartX(screenArea.getLines() / 2)
```
Comment (andreas-eberle) [inline jsettlers.main.android/src/main/java/jsettlers/main/android/core/controls/AndroidControls.java]: We use a line length of 200 characters in the project. Can you adapt this line accordingly?

File: jsettlers.main.android/src/main/java/jsettlers/main/android/core/controls/PositionChangedListener.java [hunk 4]
```diff
@@ -0,0 +1,9 @@
+package jsettlers.main.android.core.controls;
```
Comment (andreas-eberle) [inline jsettlers.main.android/src/main/java/jsettlers/main/android/core/controls/PositionChangedListener.java]: Please add the copyright notice in the new files

File: jsettlers.main.android/src/main/java/jsettlers/main/android/core/controls/PositionControls.java [hunk 5]
```diff
@@ -0,0 +1,16 @@
+package jsettlers.main.android.core.controls;
+
+import jsettlers.common.map.partition.IPartitionData;
+
+/**
+ * Created by tompr on 27/05/2017.
+ */
+
+public interface PositionControls {
+    boolean isInPlayerPartition();
```
Comment (andreas-eberle) [inline jsettlers.main.android/src/main/java/jsettlers/main/android/core/controls/PositionControls.java]: add a new line after this

File: jsettlers.main.android/src/main/java/jsettlers/main/android/gameplay/presenters/Building.java [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+import jsettlers.common.buildings.EBuildingType;
+import jsettlers.common.map.partition.IBuildingCounts;
+import jsettlers.graphics.localization.Labels;
+
+/**
+ * Created by tompr on 29/05/2017.
+ */
+
+public class Building {
```
Comment (andreas-eberle) [inline jsettlers.main.android/src/main/java/jsettlers/main/android/gameplay/presenters/Building.java]: Can you name this class so it is easily distinguished from the Building class in the game logic? Maybe something like UIBuildingInformation?

**Comment by andreas-eberle** [comment]
Thanks already @Tom-Pratt. This looks very nice already! :)

**Pull request merged**